gsuite_base/eg_common_api.e

```diff
@@ -0,0 +1,359 @@
+note
+	description: "Common abstraction to access (read/write Google API data)"
+	date: "$Date$"
+	revision: "$Revision$"
+
+deferred class
+	EG_COMMON_API
+
+inherit
+
+	LOGGABLE
+
+feature -- Reset
+
+	reset
+		do
+			create access_token.make_empty
+		end
+
+feature -- Access
+
+	access_token: STRING_32
+			-- Google OAuth2 access token.
+
+	http_status: INTEGER
+		-- Contains the last HTTP status code returned.
+
+	last_api_call: detachable STRING
+		-- Contains the last API call.
+
+	last_response: detachable OAUTH_RESPONSE
+		-- Cointains the last API response.
+
+	version: STRING_8
+			-- Google API version.
+
+	scopes: LIST [STRING_8]
+			--Lists of Authorization Scopes.
+
+feature -- Scopes
+
+	add_scope (a_scope: STRING_8)
+			-- Add an scope `a_scope` to the list of scopes.
+		do
+			scopes.force (a_scope)
+		end
+
+	clear_scopes
+			-- Remove all items.
+		do
+			scopes.wipe_out
+		ensure
+			scopes.is_empty
+		end
+
+feature -- Query Parameters Factory
+
+	query_parameters (a_params: detachable STRING_TABLE [STRING] ): detachable ARRAY [detachable TUPLE [name: STRING; value: STRING]]
+			-- @JV please add a call example
+		local
+			l_result: like query_parameters
+			l_tuple : like query_parameters.item
+			i: INTEGER
+		do
+			if attached a_params then
+				i := 1
+				create l_result.make_filled (Void, 1, a_params.count)
+				across a_params as ic
+				loop
+					create l_tuple.default_create
+					l_tuple.put (ic.key, 1)
+					l_tuple.put (ic.item, 2)
+					l_result.force (l_tuple, i)
+					i := i + 1
+				end
+				Result := l_result
+			end
+		end
+
+feature -- Error Report
+
```

Comment:
Is this only checking that the JSON returned from Google Calendar API are correct JSON?